src/coreclr/src/vm/object.cpp

```diff
@@ -210,20 +194,16 @@ TypeHandle Object::GetGCSafeTypeHandleIfPossible() const
     MethodTable * pMTToCheck = pMT;
     if (pMTToCheck->IsArray())
     {
```

Comment:
This entire special case is no longer necessary.

Reply:
What makes it different? Is that because TypeDesc was not preventing element type unload and MethodTable prevents that?

If that is the case, I am guessing the whole `GetGCSafeTypeHandleIfPossible` can be removed.

Reply:
Its different as ArrayTypeDesc::GetLoaderModule could walk through partially unloaded data structures. However, MethodTable::GetLoaderModule doesn't have that problem.

We still need GetGCSafeTypeHandleIfPossible to handle the collectiable assemblies cases. 

---
In reply to: [362658395](https://github.com/dotnet/runtime/pull/1201#discussion_r362658395) [](ancestors = 362658395)